pkg/registry/generic/registry/store_test.go

```diff
@@ -1091,17 +1130,19 @@ func TestStoreWatch(t *testing.T) {
 			}
 			wi.Stop()
 		}
-
+		destroyFunc()
 		server.Terminate(t)
 	}
 }
 
-func newTestGenericStoreRegistry(t *testing.T, hasCacheEnabled bool) (*etcdtesting.EtcdTestServer, *Store) {
+func newTestGenericStoreRegistry(t *testing.T, hasCacheEnabled bool) (*etcdtesting.EtcdTestServer, func(), *Store) {
 	podPrefix := "/pods"
 	server := etcdtesting.NewEtcdTestClientServer(t)
 	strategy := &testRESTStrategy{api.Scheme, api.SimpleNameGenerator, true, false, true}
+
 	codec := testapi.Default.StorageCodec()
 	s := etcdstorage.NewEtcdStorage(server.Client, codec, etcdtest.PathPrefix(), false, etcdtest.DeserializationCacheSize)
+	destroyFunc := func() {}
```

Comment:
In testing this change... the leak from all the tests came from the line above this and we're still leaking the connection.